diff --git a/panels/meson.build b/panels/meson.build
index 070a2e2..c90020f 100644
--- a/panels/meson.build
+++ b/panels/meson.build
@@ -4,7 +4,7 @@ panels = [
   'applications',
   'background',
   'camera',
-  'color',
+#  'color',
   'datetime',
   'default-apps',
   'diagnostics',
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index 0b57138..4bb086e 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -36,7 +36,7 @@ extern GType cc_background_panel_get_type (void);
 #ifdef BUILD_BLUETOOTH
 extern GType cc_bluetooth_panel_get_type (void);
 #endif /* BUILD_BLUETOOTH */
-extern GType cc_color_panel_get_type (void);
+//extern GType cc_color_panel_get_type (void);
 extern GType cc_date_time_panel_get_type (void);
 extern GType cc_default_apps_panel_get_type (void);
 extern GType cc_display_panel_get_type (void);
@@ -113,7 +113,7 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("bluetooth",        cc_bluetooth_panel_get_type,            NULL),
 #endif
   PANEL_TYPE("camera",           cc_camera_panel_get_type,               NULL),
-  PANEL_TYPE("color",            cc_color_panel_get_type,                NULL),
+//  PANEL_TYPE("color",            cc_color_panel_get_type,                NULL),
   PANEL_TYPE("datetime",         cc_date_time_panel_get_type,            NULL),
   PANEL_TYPE("default-apps",     cc_default_apps_panel_get_type,         NULL),
   PANEL_TYPE("diagnostics",      cc_diagnostics_panel_get_type,          cc_diagnostics_panel_static_init_func),
