diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index f209a87..13c13fe 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -93,7 +93,7 @@ extern void cc_wacom_panel_static_init_func (void);
 
 static CcPanelLoaderVtable default_panels[] =
 {
-  PANEL_TYPE("applications",     cc_applications_panel_get_type,         NULL),
+  //PANEL_TYPE("applications",     cc_applications_panel_get_type,         NULL),
   PANEL_TYPE("background",       cc_background_panel_get_type,           NULL),
 #ifdef BUILD_BLUETOOTH
   PANEL_TYPE("bluetooth",        cc_bluetooth_panel_get_type,            NULL),
@@ -104,7 +104,7 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("color",            cc_color_panel_get_type,                NULL),
   PANEL_TYPE("datetime",         cc_date_time_panel_get_type,            NULL),
   //PANEL_TYPE("default-apps",     cc_default_apps_panel_get_type,         NULL),
-  PANEL_TYPE("diagnostics",      cc_diagnostics_panel_get_type,          cc_diagnostics_panel_static_init_func),
+  //PANEL_TYPE("diagnostics",      cc_diagnostics_panel_get_type,          cc_diagnostics_panel_static_init_func),
   PANEL_TYPE("display",          cc_display_panel_get_type,              NULL),
   PANEL_TYPE("info-overview",    cc_info_overview_panel_get_type,        NULL),
   PANEL_TYPE("keyboard",         cc_keyboard_panel_get_type,             NULL),
