diff --git a/panels/keyboard/cc-keyboard-shortcut-editor.c b/panels/keyboard/cc-keyboard-shortcut-editor.c
index c3ed27d..38c9631 100644
--- a/panels/keyboard/cc-keyboard-shortcut-editor.c
+++ b/panels/keyboard/cc-keyboard-shortcut-editor.c
@@ -559,6 +559,7 @@ reset_custom_clicked_cb (CcKeyboardShortcutEditor *self)
   if (self->item)
     cc_keyboard_manager_reset_shortcut (self->manager, self->item);
 
+  memset (self->custom_combo, 0, sizeof (CcKeyCombo));
   gtk_stack_set_visible_child_name (GTK_STACK (self->custom_shortcut_stack), "button");
   gtk_widget_hide (self->reset_custom_button);
 }
@@ -635,7 +636,7 @@ setup_keyboard_item (CcKeyboardShortcutEditor *self,
 
   /* Accelerator labels */
   //gtk_shortcut_label_set_accelerator (GTK_SHORTCUT_LABEL (self->shortcut_accel_label), accel);
-  //gtk_shortcut_label_set_accelerator (GTK_SHORTCUT_LABEL (self->custom_shortcut_accel_label), accel);
+  gtk_shortcut_label_set_accelerator (GTK_SHORTCUT_LABEL (self->custom_shortcut_accel_label), accel);
 
   /* Setup the custom entries */
   if (is_custom)
