diff --git a/panels/background/cc-background-chooser.c b/panels/background/cc-background-chooser.c
index 9843379..6cb82df 100644
--- a/panels/background/cc-background-chooser.c
+++ b/panels/background/cc-background-chooser.c
@@ -97,7 +97,7 @@ on_file_dialog_response_cb (GtkDialog           *dialog,
                             gint                 response,
                             CcBackgroundChooser *chooser)
 {
-  gboolean result;
+  gboolean result = TRUE;
 
   if (response == GTK_RESPONSE_ACCEPT) {
     g_autofree gchar *filename = NULL;
