diff --git a/panels/network/cc-network-panel.c b/panels/network/cc-network-panel.c
index 2651edc..b909f5f 100644
--- a/panels/network/cc-network-panel.c
+++ b/panels/network/cc-network-panel.c
@@ -238,6 +238,8 @@ cc_network_panel_get_help_uri (CcPanel *panel)
 	return "help:gnome-help/net";
 }
 
+static void update_vpn_section (CcNetworkPanel *self);
+
 static void
 object_removed_cb (NetObject *object, CcNetworkPanel *panel)
 {
@@ -247,6 +249,10 @@ object_removed_cb (NetObject *object, CcNetworkPanel *panel)
         stack = g_hash_table_lookup (panel->device_to_stack, object);
         if (stack != NULL)
                 gtk_widget_destroy (stack);
+
+        panel->devices->len--;
+
+        update_vpn_section(panel);
 }
 
 GPtrArray *
