From 59989751bc5c34da9268b0178d2f3a5810b01883 Mon Sep 17 00:00:00 2001
From: HaeRyong Jeong <jeong89@gooroom.kr>
Date: Thu, 25 Jun 2020 19:13:13 +0900
Subject: [PATCH] Disabled speaker test in sound panel

---
 panels/sound/gvc-mixer-dialog.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/sound/gvc-mixer-dialog.c b/panels/sound/gvc-mixer-dialog.c
index 5e6085f..5034a45 100644
--- a/panels/sound/gvc-mixer-dialog.c
+++ b/panels/sound/gvc-mixer-dialog.c
@@ -254,7 +254,7 @@ update_output_settings (GvcMixerDialog      *dialog,
 
                 g_object_set (G_OBJECT (dialog->output_profile_combo), "button-label", _("_Test Speakers"), NULL);
                 g_object_set (G_OBJECT (dialog->output_profile_combo),
-                              "show-button", TRUE, NULL);
+                              "show-button", FALSE, NULL);
                 g_signal_connect (G_OBJECT (dialog->output_profile_combo), "button-clicked",
                                   G_CALLBACK (on_test_speakers_clicked), dialog);
 
-- 
2.20.1

