diff --git a/panels/font/font-chooser-widget.ui b/panels/font/font-chooser-widget.ui
index a52d4ee..e0b1167 100644
--- a/panels/font/font-chooser-widget.ui
+++ b/panels/font/font-chooser-widget.ui
@@ -53,7 +53,7 @@
                 <property name="placeholder-text" translatable="yes">Search font name</property>
                 <signal name="search-changed" handler="text_changed_cb" swapped="no"/>
                 <signal name="stop-search" handler="stop_search_cb" swapped="no"/>
-                <signal name="preedit-search" handler="preedit_search_cb" swapped="no"/>
+                <signal name="preedit-changed" handler="preedit_changed_cb" swapped="no"/>
               </object>
               <packing>
                 <property name="left-attach">0</property>
