diff --git a/panels/network/net-device-wifi.c b/panels/network/net-device-wifi.c
index ce9a4be..c58d22d 100644
--- a/panels/network/net-device-wifi.c
+++ b/panels/network/net-device-wifi.c
@@ -2051,6 +2051,42 @@ show_details_for_row (GtkButton *button, NetDeviceWifi *device_wifi)
         net_connection_editor_run (editor);
 }
 
+GtkWidget *
+create_label (void)
+{
+        GtkWidget *label;
+        const char *str = _("There are no known Wi-Fi Networks.");
+        const char *format = "<span font_desc=\"15.0\">\%s</span>";
+        char *markup;
+        
+        label = gtk_label_new (NULL);
+        markup = g_markup_printf_escaped (format, str);
+        gtk_label_set_markup (GTK_LABEL (label), markup);
+        gtk_widget_set_margin_top (GTK_WIDGET (label), 10);
+        gtk_widget_set_margin_bottom (GTK_WIDGET (label), 10);
+        g_free (markup);
+        return label;
+}
+
+static void
+update_list (GtkContainer *list, GtkWidget *row, NetDeviceWifi *device_wifi)
+{
+        GSList *connections;
+        GtkWidget *label;
+        guint length;
+
+        if (device_wifi != NULL) {
+                connections = net_device_get_valid_connections (NET_DEVICE (device_wifi));
+
+                length = g_list_length (connections);
+                if (length == 2) {
+                        label = create_label();
+                        gtk_widget_show (label);
+                        gtk_container_add (GTK_CONTAINER (list), label);
+                }
+        }
+}
+
 static void
 open_history (NetDeviceWifi *device_wifi)
 {
@@ -2069,7 +2105,9 @@ open_history (NetDeviceWifi *device_wifi)
         GPtrArray *aps_unique = NULL;
         NMAccessPoint *active_ap;
         guint i;
+        guint length;
         NMDevice *nm_device;
+        GtkWidget *label;
         GtkWidget *list;
         GtkWidget *row;
         GtkSizeGroup *rows;
@@ -2169,12 +2207,22 @@ open_history (NetDeviceWifi *device_wifi)
 
                 make_row (rows, icons, forget, nm_device, connection, ap, active_ap, &row, NULL, &button);
                 gtk_container_add (GTK_CONTAINER (list), row);
+                g_signal_connect (GTK_CONTAINER (list), "remove",
+                                  G_CALLBACK (update_list), device_wifi);
                 if (button) {
                         g_signal_connect (button, "clicked",
                                           G_CALLBACK (show_details_for_row), device_wifi);
                         g_object_set_data (G_OBJECT (button), "row", row);
                 }
         }
+
+        length = g_slist_length (connections);
+        if (length == 1) {
+                label = create_label();
+                gtk_widget_show (label);
+                gtk_container_add (GTK_CONTAINER (list), label);
+        }
+
         g_slist_free (connections);
         g_ptr_array_free (aps_unique, TRUE);
 
diff --git a/po/ko.po b/po/ko.po
index c47ae7d..a142b1c 100644
--- a/po/ko.po
+++ b/po/ko.po
@@ -2887,12 +2887,16 @@ msgstr ""
 msgid "_Forget"
 msgstr "저장 지우기(_F)"
 
-#: panels/network/net-device-wifi.c:2082 panels/network/net-device-wifi.c:2089
+#: panels/network/net-device-wifi.c:2058 
+msgid "There are no known Wi-Fi Networks."
+msgstr "알려진 와이파이 네트워크가 없습니다."
+
+#: panels/network/net-device-wifi.c:2120 panels/network/net-device-wifi.c:2127
 msgid "Known Wi-Fi Networks"
 msgstr "알려진 와이파이 네트워크"
 
 #. translators: This is the label for the "Forget wireless network" functionality
-#: panels/network/net-device-wifi.c:2122
+#: panels/network/net-device-wifi.c:2160
 msgctxt "Wi-Fi Network"
 msgid "_Forget"
 msgstr "지우기(_F)"
