diff --git a/panels/printers/cc-printers-panel.c b/panels/printers/cc-printers-panel.c
index 68f0348..13e15a3 100644
--- a/panels/printers/cc-printers-panel.c
+++ b/panels/printers/cc-printers-panel.c
@@ -1387,13 +1387,15 @@ filter_function (GtkListBoxRow *row,
   g_free (printer_name);
   g_free (printer_location);
 
-  search = cc_util_normalize_casefold_and_unaccent (priv->filter_text);
-
-  retval = strstr (name, search) != NULL;
-  if (location != NULL)
-      retval = retval || (strstr (location, search) != NULL);
-
-  g_free (search);
+  if (priv->filter_text) {
+    search = cc_util_normalize_casefold_and_unaccent (priv->filter_text);
+  
+    retval = strstr (name, search) != NULL;
+    if (location != NULL)
+        retval = retval || (strstr (location, search) != NULL);
+  
+    g_free (search);
+  }
   g_free (name);
   g_free (location);
 
