From 0c20ff2745cb711ea9be13df0ced8bd3570845ed Mon Sep 17 00:00:00 2001
From: HaeRyong Jeong <jeong89@gooroom.kr>
Date: Wed, 24 Jun 2020 14:58:32 +0900
Subject: [PATCH] Fixed search entry completion in datetime

---
 panels/datetime/cc-datetime-panel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/datetime/cc-datetime-panel.c b/panels/datetime/cc-datetime-panel.c
index 51d5fc8..1f9c8f7 100644
--- a/panels/datetime/cc-datetime-panel.c
+++ b/panels/datetime/cc-datetime-panel.c
@@ -468,7 +468,7 @@ city_changed_cb (GtkEntryCompletion *entry_completion,
   cc_timezone_map_set_timezone (CC_TIMEZONE_MAP (self->map), zone);
 
   entry = gtk_entry_completion_get_entry (GTK_ENTRY_COMPLETION (entry_completion));
-  gtk_entry_set_text (GTK_ENTRY (entry), "");
+  //gtk_entry_set_text (GTK_ENTRY (entry), "");
 
   return TRUE;
 }
-- 
2.20.1

