diff --git a/panels/display/cc-display-panel.c b/panels/display/cc-display-panel.c
index f95b0b4..ca61e85 100644
--- a/panels/display/cc-display-panel.c
+++ b/panels/display/cc-display-panel.c
@@ -219,7 +219,7 @@ config_ensure_of_type (CcDisplayPanel *panel, CcDisplayConfigType type)
           CcDisplayMonitor *output = l->data;
 
           cc_display_monitor_set_active (output, cc_display_monitor_is_usable (output));
-          cc_display_monitor_set_mode (output, cc_display_monitor_get_preferred_mode (output));
+          cc_display_monitor_set_mode (output, cc_display_monitor_get_mode (output));
         }
       break;
 
diff --git a/po/ko.po b/po/ko.po
index 84d5a18..82bdedf 100644
--- a/po/ko.po
+++ b/po/ko.po
@@ -1622,7 +1622,7 @@ msgstr "단일 디스플레이"
 #: panels/display/cc-display-panel.ui:108
 #: panels/display/cc-display-panel.ui:310
 msgid "Join Displays"
-msgstr "디스플레이 연결"
+msgstr "디스플레이 확장"
 
 #: panels/display/cc-display-panel.ui:126
 msgid "Mirror"
