diff --git a/panels/universal-access/cc-ua-panel.c b/panels/universal-access/cc-ua-panel.c
index d7aac25..0fadb34 100644
--- a/panels/universal-access/cc-ua-panel.c
+++ b/panels/universal-access/cc-ua-panel.c
@@ -127,6 +127,7 @@ struct _CcUaPanel
   GtkWidget *repeat_keys_speed_scale;
   GtkWidget *repeat_keys_switch;
   GtkWidget *row_accessx;
+  GtkWidget *row_locate_pointer_assist;
   GtkWidget *row_click_assist;
   //GtkWidget *row_cursor_blinking;
   GtkWidget *row_cursor_size;
@@ -263,6 +264,7 @@ cc_ua_panel_class_init (CcUaPanelClass *klass)
   gtk_widget_class_bind_template_child (widget_class, CcUaPanel, repeat_keys_speed_scale);
   gtk_widget_class_bind_template_child (widget_class, CcUaPanel, repeat_keys_switch);
   gtk_widget_class_bind_template_child (widget_class, CcUaPanel, row_accessx);
+  gtk_widget_class_bind_template_child (widget_class, CcUaPanel, row_locate_pointer_assist);
   gtk_widget_class_bind_template_child (widget_class, CcUaPanel, row_click_assist);
   //gtk_widget_class_bind_template_child (widget_class, CcUaPanel, row_cursor_blinking);
   gtk_widget_class_bind_template_child (widget_class, CcUaPanel, row_cursor_size);
@@ -1010,6 +1012,9 @@ cc_ua_panel_init_mouse (CcUaPanel *self)
     gtk_widget_set_visible (self->row_click_assist, FALSE);
   }
 
+  if (!g_file_test ("/usr/lib/gnome-shell", G_FILE_TEST_EXISTS))
+    gtk_widget_set_visible (self->row_locate_pointer_assist, FALSE);
+
   add_separators (GTK_LIST_BOX (list));
 
   g_signal_connect_object (list, "row-activated",
diff --git a/panels/universal-access/cc-ua-panel.ui b/panels/universal-access/cc-ua-panel.ui
index c51b0ee..738f220 100644
--- a/panels/universal-access/cc-ua-panel.ui
+++ b/panels/universal-access/cc-ua-panel.ui
@@ -795,7 +795,7 @@
                                       </object>
                                     </child>
                                     <child>
-                                      <object class="GtkListBoxRow">
+                                      <object class="GtkListBoxRow" id="row_locate_pointer_assist">
                                         <property name="visible">True</property>
                                         <property name="can_focus">True</property>
                                         <property name="selectable">False</property>
