diff --git a/panels/meson.build b/panels/meson.build
index fa11379..7a12349 100644
--- a/panels/meson.build
+++ b/panels/meson.build
@@ -6,7 +6,7 @@ panels = [
   'camera',
   'color',
   'datetime',
-  'default-apps',
+#  'default-apps',
   'diagnostics',
   'display',
   'info-overview',
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index 3ab5aca..f209a87 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -39,7 +39,7 @@ extern GType cc_bluetooth_panel_get_type (void);
 #endif /* BUILD_BLUETOOTH */
 extern GType cc_color_panel_get_type (void);
 extern GType cc_date_time_panel_get_type (void);
-extern GType cc_default_apps_panel_get_type (void);
+//extern GType cc_default_apps_panel_get_type (void);
 extern GType cc_display_panel_get_type (void);
 extern GType cc_info_overview_panel_get_type (void);
 extern GType cc_keyboard_panel_get_type (void);
@@ -103,7 +103,7 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("camera",           cc_camera_panel_get_type,               NULL),
   PANEL_TYPE("color",            cc_color_panel_get_type,                NULL),
   PANEL_TYPE("datetime",         cc_date_time_panel_get_type,            NULL),
-  PANEL_TYPE("default-apps",     cc_default_apps_panel_get_type,         NULL),
+  //PANEL_TYPE("default-apps",     cc_default_apps_panel_get_type,         NULL),
   PANEL_TYPE("diagnostics",      cc_diagnostics_panel_get_type,          cc_diagnostics_panel_static_init_func),
   PANEL_TYPE("display",          cc_display_panel_get_type,              NULL),
   PANEL_TYPE("info-overview",    cc_info_overview_panel_get_type,        NULL),
diff --git a/shell/org.gnome.ControlCenter.gschema.xml b/shell/org.gnome.ControlCenter.gschema.xml
index d9178b6..fe4854d 100644
--- a/shell/org.gnome.ControlCenter.gschema.xml
+++ b/shell/org.gnome.ControlCenter.gschema.xml
@@ -16,7 +16,7 @@
       </description>
     </key>
     <key name="whitelist-panels" type="as">
-      <default>['bluetooth', 'background', 'region', 'universal-access', 'font', 'privacy', 'location', 'camera', 'microphone', 'usage', 'lock', 'sound', 'power', 'network', 'wifi', 'wacom', 'display', 'keyboard', 'printers', 'mouse', 'removable-media', 'thunderbolt', 'color', 'info-overview', 'datetime', 'user-accounts', 'default-apps', 'themes']</default>
+      <default>['bluetooth', 'background', 'region', 'universal-access', 'font', 'privacy', 'location', 'camera', 'microphone', 'usage', 'lock', 'sound', 'power', 'network', 'wifi', 'wacom', 'display', 'keyboard', 'printers', 'mouse', 'removable-media', 'thunderbolt', 'color', 'info-overview', 'datetime', 'user-accounts', 'themes']</default>
       <summary>Show settings list</summary>
       <description>Show side panel list that matches the User settings values.</description>
     </key>
