diff --git a/panels/user-accounts/cc-add-user-dialog.c b/panels/user-accounts/cc-add-user-dialog.c
index bbbe851..0a26561 100644
--- a/panels/user-accounts/cc-add-user-dialog.c
+++ b/panels/user-accounts/cc-add-user-dialog.c
@@ -131,7 +131,7 @@ show_error_dialog (CcAddUserDialog *self,
                 g_dbus_error_strip_remote_error (error);
                 gtk_window_set_title (GTK_WINDOW (dialog), _("Error"));
                 gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog),
-                                                          "%s", _("Username cannot start with digits."));
+                                                          "%s", _("Username starts with number or invalid character is included such as special character or korean."));
         }
 
         g_signal_connect (dialog, "response", G_CALLBACK (gtk_widget_destroy), NULL);
diff --git a/po/ko.po b/po/ko.po
index 50c0312..4bc2c7b 100644
--- a/po/ko.po
+++ b/po/ko.po
@@ -7031,8 +7031,8 @@ msgid "Error"
 msgstr "에러"
 
 #: panels/user-accounts/um-account-dialog.c:134
-msgid "Username cannot start with digits."
-msgstr "사용자 이름은 숫자로 시작할 수 없습니다."
+msgid "Username starts with number or invalid character is included such as special character or korean."
+msgstr "사용자 이름이 숫자로 시작하거나 특수문자 또는 한국어와 같은 유효하지 않은 문자가 포함되어 있습니다."
 
 #: panels/user-accounts/cc-add-user-dialog.c:205
 msgid "Failed to add account"
