diff --git a/panels/universal-access/uap.ui b/panels/universal-access/uap.ui
index 128ad6a..c8eca10 100644
--- a/panels/universal-access/uap.ui
+++ b/panels/universal-access/uap.ui
@@ -17,7 +17,7 @@
     <property name="page_increment">10</property>
   </object>
   <object class="GtkAdjustment" id="repeat_speed_adjustment">
-    <property name="lower">0.5</property>
+    <property name="lower">10</property>
     <property name="upper">110</property>
     <property name="value">33.3</property>
     <property name="step_increment">1</property>
