From 141e51bc0581f4ced15033a386a77d15d5c406ac Mon Sep 17 00:00:00 2001
From: "boyeon.choi" <boyeon.choi@gooroom.kr>
Date: Mon, 12 Apr 2021 14:32:49 +0900
Subject: [PATCH] Enable F1 key for yelp

---
 shell/cc-application.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/shell/cc-application.c b/shell/cc-application.c
index fb15a64..27bf5d1 100644
--- a/shell/cc-application.c
+++ b/shell/cc-application.c
@@ -230,7 +230,7 @@ cc_application_startup (GApplication *application)
   CcApplication *self = CC_APPLICATION (application);
   //GMenu *section;
   //GMenu *menu;
-  //const gchar *help_accels[] = { "F1", NULL };
+  const gchar *help_accels[] = { "F1", NULL };
 
   g_action_map_add_action_entries (G_ACTION_MAP (self),
                                    cc_app_actions,
@@ -259,8 +259,8 @@ cc_application_startup (GApplication *application)
   //gtk_application_set_app_menu (GTK_APPLICATION (application),
   //                              G_MENU_MODEL (menu));
 
-  //gtk_application_set_accels_for_action (GTK_APPLICATION (application),
-  //                                       "app.help", help_accels);
+  gtk_application_set_accels_for_action (GTK_APPLICATION (application),
+                                         "app.help", help_accels);
 
   self->model = cc_shell_model_new ();
   self->window = cc_window_new (GTK_APPLICATION (application), self->model);
-- 
2.20.1

