From 5e94d1bf19aa56d92dc90e01ed4dd5f49a415eb7 Mon Sep 17 00:00:00 2001
From: HaeRyong Jeong <jeong89@gooroom.kr>
Date: Fri, 3 Jul 2020 18:55:50 +0900
Subject: [PATCH] Change default background style to zoom

---
 panels/background/bg-colors-source.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/background/bg-colors-source.c b/panels/background/bg-colors-source.c
index 7c837b8..95725b1 100644
--- a/panels/background/bg-colors-source.c
+++ b/panels/background/bg-colors-source.c
@@ -100,7 +100,7 @@ bg_colors_source_add_color (BgColorsSource               *self,
                 "primary-color", color,
                 "secondary-color", color,
                 "shading", G_DESKTOP_BACKGROUND_SHADING_SOLID,
-                "placement", G_DESKTOP_BACKGROUND_STYLE_WALLPAPER,
+                "placement", G_DESKTOP_BACKGROUND_STYLE_ZOOM,
                 "flags", flags,
                 NULL);
   cc_background_item_load (item, NULL);
-- 
2.20.1

