diff --git a/panels/printers/pp-options-dialog.ui b/panels/printers/pp-options-dialog.ui
index 79eaf88..a003fb5 100644
--- a/panels/printers/pp-options-dialog.ui
+++ b/panels/printers/pp-options-dialog.ui
@@ -7,7 +7,7 @@
     <property name="height_request">400</property>
     <property name="can_focus">False</property>
     <property name="border_width">5</property>
-    <property name="resizable">False</property>
+    <property name="resizable">True</property>
     <property name="type_hint">dialog</property>
     <child internal-child="headerbar">
       <object class="GtkHeaderBar">
