diff --git a/panels/info/info.gresource.xml b/panels/info/info.gresource.xml
index c967223..d7e349b 100644
--- a/panels/info/info.gresource.xml
+++ b/panels/info/info.gresource.xml
@@ -2,7 +2,7 @@
 <gresources>
   <gresource prefix="/org/gnome/control-center/info">
     <file preprocess="xml-stripblanks">info-overview.ui</file>
-    <file preprocess="xml-stripblanks">info-default-apps.ui</file>
+    <!--file preprocess="xml-stripblanks">info-default-apps.ui</file-->
     <file preprocess="xml-stripblanks">info-removable-media.ui</file>
     <file>GnomeLogoVerticalMedium.svg</file>
   </gresource>
diff --git a/panels/info/meson.build b/panels/info/meson.build
index dc1c11c..064710c 100644
--- a/panels/info/meson.build
+++ b/panels/info/meson.build
@@ -46,7 +46,7 @@ sources = files(gsd_sources) + files(
 
 resource_data = files(
   'GnomeLogoVerticalMedium.svg',
-  'info-default-apps.ui',
+  #'info-default-apps.ui',
   'info-overview.ui',
   'info-removable-media.ui'
 )
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index e24a38c..0c30d1d 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -39,7 +39,7 @@ extern GType cc_color_panel_get_type (void);
 extern GType cc_date_time_panel_get_type (void);
 extern GType cc_display_panel_get_type (void);
 extern GType cc_info_overview_panel_get_type (void);
-extern GType cc_info_default_apps_panel_get_type (void);
+//extern GType cc_info_default_apps_panel_get_type (void);
 extern GType cc_info_removable_media_panel_get_type (void);
 extern GType cc_keyboard_panel_get_type (void);
 extern GType cc_mouse_panel_get_type (void);
@@ -100,7 +100,7 @@ static struct {
   PANEL_TYPE("datetime",         cc_date_time_panel_get_type,            NULL),
   PANEL_TYPE("display",          cc_display_panel_get_type,              NULL),
   PANEL_TYPE("info-overview",    cc_info_overview_panel_get_type,        NULL),
-  PANEL_TYPE("default-apps",     cc_info_default_apps_panel_get_type,    NULL),
+//  PANEL_TYPE("default-apps",     cc_info_default_apps_panel_get_type,    NULL),
   PANEL_TYPE("removable-media",  cc_info_removable_media_panel_get_type, NULL),
   PANEL_TYPE("keyboard",         cc_keyboard_panel_get_type,             NULL),
   PANEL_TYPE("mouse",            cc_mouse_panel_get_type,                NULL),
