diff --git a/panels/user-accounts/cc-add-user-dialog.c b/panels/user-accounts/cc-add-user-dialog.c
index 201ac41..23399ab 100644
--- a/panels/user-accounts/cc-add-user-dialog.c
+++ b/panels/user-accounts/cc-add-user-dialog.c
@@ -762,6 +762,16 @@ local_password_entry_changed_cb (CcAddUserDialog *self)
         recheck_password_match (self);
 }
 
+static void
+clear_verify_entry (CcAddUserDialog *self)
+{
+       if (strlen (gtk_entry_get_text (self->local_password_entry)) !=
+           strlen (gtk_entry_get_text (self->local_verify_entry)))
+       {
+               gtk_entry_set_text (self->local_verify_entry, "");
+       }
+}
+
 static void
 local_verify_entry_changed_cb (CcAddUserDialog *self)
 {
@@ -1742,6 +1752,7 @@ cc_add_user_dialog_class_init (CcAddUserDialogClass *klass)
         gtk_widget_class_bind_template_callback (widget_class, local_name_entry_changed_cb);
         gtk_widget_class_bind_template_callback (widget_class, local_name_entry_focus_out_event_cb);
         gtk_widget_class_bind_template_callback (widget_class, local_password_entry_changed_cb);
+        gtk_widget_class_bind_template_callback (widget_class, clear_verify_entry);
         gtk_widget_class_bind_template_callback (widget_class, local_password_entry_icon_press_cb);
         gtk_widget_class_bind_template_callback (widget_class, local_password_entry_key_press_event_cb);
         gtk_widget_class_bind_template_callback (widget_class, local_password_radio_changed_cb);
diff --git a/panels/user-accounts/cc-add-user-dialog.ui b/panels/user-accounts/cc-add-user-dialog.ui
index 12a253a..e2d87d5 100644
--- a/panels/user-accounts/cc-add-user-dialog.ui
+++ b/panels/user-accounts/cc-add-user-dialog.ui
@@ -335,6 +335,7 @@
                     <property name="activates_default">True</property>
                     <property name="input_purpose">password</property>
                     <signal name="notify::text" handler="local_password_entry_changed_cb" object="CcAddUserDialog" swapped="yes"/>
+                    <signal name="notify::text" handler="clear_verify_entry" object="CcAddUserDialog" swapped="yes"/>
                     <signal name="icon-press" handler="local_password_entry_icon_press_cb" object="CcAddUserDialog" swapped="yes"/>
                     <signal name="focus-out-event" handler="password_focus_out_event_cb" after="yes" object="CcAddUserDialog" swapped="yes"/>
                     <signal name="key-press-event" handler="local_password_entry_key_press_event_cb" object="CcAddUserDialog" swapped="yes"/>
