diff --git a/panels/user-accounts/um-password-dialog.c b/panels/user-accounts/um-password-dialog.c
index 0ea1e71..428379c 100644
--- a/panels/user-accounts/um-password-dialog.c
+++ b/panels/user-accounts/um-password-dialog.c
@@ -430,6 +430,18 @@ old_password_entry_changed (GtkEntry         *entry,
                                                            um);
 }
 
+static void
+clear_verify_entry (GtkEntry		*entry,
+		    GParamSpec		*pspec,
+		    UmPasswordDialog	*um)
+{
+	if (strlen (gtk_entry_get_text (GTK_ENTRY (um->password_entry))) !=
+	    strlen (gtk_entry_get_text (GTK_ENTRY (um->verify_entry))))
+	{
+		gtk_entry_set_text (GTK_ENTRY (um->verify_entry), "");
+	}
+}
+
 static void
 on_generate (GtkEntry             *entry,
              GtkEntryIconPosition  pos,
@@ -493,6 +505,8 @@ um_password_dialog_new (void)
         widget = (GtkWidget *) gtk_builder_get_object (builder, "password-entry");
         g_signal_connect (widget, "notify::text",
                           G_CALLBACK (password_entry_changed), um);
+	g_signal_connect (widget, "notify::text",
+			  G_CALLBACK (clear_verify_entry), um);
         g_signal_connect_after (widget, "focus-out-event",
                                 G_CALLBACK (password_entry_focus_out), um);
         g_signal_connect (widget, "key-press-event",
