diff --git a/panels/user-accounts/cc-add-user-dialog.c b/panels/user-accounts/cc-add-user-dialog.c
index cb584d9..70a37f9 100644
--- a/panels/user-accounts/cc-add-user-dialog.c
+++ b/panels/user-accounts/cc-add-user-dialog.c
@@ -75,6 +75,7 @@ struct _CcAddUserDialog {
         GtkComboBoxText    *local_username_combo;
         GtkListStore       *local_username_model;
         GtkEntry           *local_password_entry;
+        GtkRadioButton     *local_password_login_radio;
         GtkRadioButton     *local_password_radio;
         GtkEntry           *local_username_entry;
         GtkLabel           *local_username_hint_label;
@@ -801,6 +802,19 @@ local_password_radio_changed_cb (CcAddUserDialog *self)
         dialog_validate (self);
 }
 
+static void
+account_type_admin_changed_cb (CcAddUserDialog *self)
+{
+        gboolean active;
+
+        active = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (self->local_account_type_standard));
+
+        gtk_widget_set_sensitive (GTK_WIDGET (self->local_password_login_radio), active);
+        if (!active) {
+                g_signal_emit_by_name (self->local_password_radio, "clicked", 0);
+        }
+}
+
 static gboolean
 enterprise_validate (CcAddUserDialog *self)
 {
@@ -1775,6 +1789,7 @@ cc_add_user_dialog_class_init (CcAddUserDialogClass *klass)
         gtk_widget_class_bind_template_child (widget_class, CcAddUserDialog, local_username_combo);
         gtk_widget_class_bind_template_child (widget_class, CcAddUserDialog, local_username_model);
         gtk_widget_class_bind_template_child (widget_class, CcAddUserDialog, local_password_entry);
+        gtk_widget_class_bind_template_child (widget_class, CcAddUserDialog, local_password_login_radio);
         gtk_widget_class_bind_template_child (widget_class, CcAddUserDialog, local_password_radio);
         gtk_widget_class_bind_template_child (widget_class, CcAddUserDialog, local_username_entry);
         gtk_widget_class_bind_template_child (widget_class, CcAddUserDialog, local_username_hint_label);
@@ -1804,6 +1819,7 @@ cc_add_user_dialog_class_init (CcAddUserDialogClass *klass)
         gtk_widget_class_bind_template_callback (widget_class, local_username_combo_focus_out_event_cb);
         gtk_widget_class_bind_template_callback (widget_class, local_verify_entry_changed_cb);
         gtk_widget_class_bind_template_callback (widget_class, password_focus_out_event_cb);
+        gtk_widget_class_bind_template_callback (widget_class, account_type_admin_changed_cb);
 }
 
 CcAddUserDialog *
diff --git a/panels/user-accounts/cc-add-user-dialog.ui b/panels/user-accounts/cc-add-user-dialog.ui
index 1298145..09e423f 100644
--- a/panels/user-accounts/cc-add-user-dialog.ui
+++ b/panels/user-accounts/cc-add-user-dialog.ui
@@ -232,6 +232,7 @@
                         <property name="draw_indicator">False</property>
                         <property name="height_request">35</property>
                         <property name="group">local_account_type_standard</property>
+                        <signal name="toggled" handler="account_type_admin_changed_cb" object="CcAddUserDialog" swapped="yes"/>
                       </object>
                     </child>
                   </object>
