diff --git a/panels/power/cc-power-panel.c b/panels/power/cc-power-panel.c
index 6e283f9..094886c 100644
--- a/panels/power/cc-power-panel.c
+++ b/panels/power/cc-power-panel.c
@@ -972,6 +972,7 @@ sync_kbd_brightness (CcPowerPanel *self)
       /* set the slider */
       brightness = g_variant_get_int32 (result);
       visible = brightness >= 0.0;
+      brightness = brightness > 0 && brightness < 100 ? brightness + 1 : brightness;
     }
   else
     {
@@ -983,8 +984,6 @@ sync_kbd_brightness (CcPowerPanel *self)
   if (visible)
     {
       range = GTK_RANGE (self->kbd_brightness_scale);
-      gtk_range_set_range (range, 0, 100);
-      gtk_range_set_increments (range, 1, 10);
       self->kbd_setting_brightness = TRUE;
       gtk_range_set_value (range, brightness);
       self->kbd_setting_brightness = FALSE;
@@ -1006,6 +1005,7 @@ sync_screen_brightness (CcPowerPanel *self)
       /* set the slider */
       brightness = g_variant_get_int32 (result);
       visible = brightness >= 0.0;
+      brightness = brightness > 0 && brightness < 100 ? brightness + 1 : brightness;
     }
   else
     {
@@ -1018,8 +1018,6 @@ sync_screen_brightness (CcPowerPanel *self)
   if (visible)
     {
       range = GTK_RANGE (self->brightness_scale);
-      gtk_range_set_range (range, 1, 100);
-      gtk_range_set_increments (range, 1, 10);
       self->setting_brightness = TRUE;
       gtk_range_set_value (range, brightness);
       self->setting_brightness = FALSE;
@@ -1694,7 +1692,7 @@ add_brightness_row (CcPowerPanel  *self,
   gtk_box_pack_start (GTK_BOX (box2), w, FALSE, FALSE, 0);
   gtk_size_group_add_widget (self->charge_sizegroup, w);
 
-  *brightness_scale = scale = gtk_scale_new_with_range (GTK_ORIENTATION_HORIZONTAL, 1, 100, 1);
+  *brightness_scale = scale = gtk_scale_new_with_range (GTK_ORIENTATION_HORIZONTAL, 0, 100, 1);
   //gtk_label_set_mnemonic_widget (GTK_LABEL (label), scale);
   gtk_scale_set_draw_value (GTK_SCALE (scale), TRUE);
   gtk_scale_set_value_pos (GTK_SCALE (scale), GTK_POS_BOTTOM);
