diff --git a/subprojects/gvc/gvc-mixer-control.c b/subprojects/gvc/gvc-mixer-control.c
index 8b39080..9168b5c 100644
--- a/subprojects/gvc/gvc-mixer-control.c
+++ b/subprojects/gvc/gvc-mixer-control.c
@@ -1544,6 +1544,7 @@ update_sink (GvcMixerControl    *control,
         gvc_mixer_stream_set_is_muted (stream, info->mute);
         gvc_mixer_stream_set_can_decibel (stream, !!(info->flags & PA_SINK_DECIBEL_VOLUME));
         gvc_mixer_stream_set_base_volume (stream, (guint32) info->base_volume);
+        sync_effects_base_volume_and_scale (stream);
         gvc_mixer_stream_set_state (stream, translate_pa_state (info->state));
 
         /* Messy I know but to set the port everytime regardless of whether it has changed will cost us a
@@ -1676,6 +1677,7 @@ update_source (GvcMixerControl      *control,
         gvc_mixer_stream_set_is_muted (stream, info->mute);
         gvc_mixer_stream_set_can_decibel (stream, !!(info->flags & PA_SOURCE_DECIBEL_VOLUME));
         gvc_mixer_stream_set_base_volume (stream, (guint32) info->base_volume);
+        sync_effects_base_volume_and_scale (stream);
         g_debug ("update source");
 
         if (info->active_port != NULL) {
diff --git a/subprojects/gvc/gvc-mixer-stream.c b/subprojects/gvc/gvc-mixer-stream.c
index c324900..cf1daa8 100644
--- a/subprojects/gvc/gvc-mixer-stream.c
+++ b/subprojects/gvc/gvc-mixer-stream.c
@@ -515,6 +515,15 @@ gvc_mixer_stream_set_base_volume (GvcMixerStream *stream,
         return TRUE;
 }
 
+void
+sync_effects_base_volume_and_scale (GvcMixerStream *stream)
+{
+    gboolean result = NULL;
+    if (stream->priv->base_volume == 65536) {
+        result = gvc_mixer_stream_set_base_volume (stream, 1);
+    }
+}
+
 const GvcMixerStreamPort *
 gvc_mixer_stream_get_port (GvcMixerStream *stream)
 {
diff --git a/subprojects/gvc/gvc-mixer-stream.h b/subprojects/gvc/gvc-mixer-stream.h
index 586ec75..7363859 100644
--- a/subprojects/gvc/gvc-mixer-stream.h
+++ b/subprojects/gvc/gvc-mixer-stream.h
@@ -87,6 +87,7 @@ pa_volume_t         gvc_mixer_stream_get_volume      (GvcMixerStream *stream);
 gdouble             gvc_mixer_stream_get_decibel     (GvcMixerStream *stream);
 gboolean            gvc_mixer_stream_push_volume     (GvcMixerStream *stream);
 pa_volume_t         gvc_mixer_stream_get_base_volume (GvcMixerStream *stream);
+void                sync_effects_base_volume_and_scale (GvcMixerStream *stream);
 
 gboolean            gvc_mixer_stream_get_is_muted    (GvcMixerStream *stream);
 gboolean            gvc_mixer_stream_get_can_decibel (GvcMixerStream *stream);
