diff --git a/panels/keyboard/cc-keyboard-shortcut-editor.c b/panels/keyboard/cc-keyboard-shortcut-editor.c
index 18e7871..2d54aaa 100644
--- a/panels/keyboard/cc-keyboard-shortcut-editor.c
+++ b/panels/keyboard/cc-keyboard-shortcut-editor.c
@@ -42,6 +42,7 @@ struct _CcKeyboardShortcutEditor
   GtkWidget          *reset_custom_button;
   GtkWidget          *set_button;
   GtkWidget          *shortcut_accel_label;
+  GtkWidget          *shortcut_reset_label;
   GtkWidget          *shortcut_conflict_label;
   GtkWidget          *standard_shortcut_stack;
   GtkWidget          *stack;
@@ -454,6 +455,7 @@ setup_custom_shortcut (CcKeyboardShortcutEditor *self)
       label = is_custom_shortcut (self) ? self->new_shortcut_conflict_label : self->shortcut_conflict_label;
 
       gtk_label_set_markup (GTK_LABEL (label), collision_text);
+      gtk_widget_set_visible (label, collision_item != NULL);
 
       g_free (friendly_accelerator);
       g_free (collision_text);
@@ -586,7 +588,8 @@ reset_item_clicked_cb (CcKeyboardShortcutEditor *self)
     self->custom_combo->keyval = default_combo->keyval;
     self->custom_combo->mask = default_combo->mask;
   }
-
+  gtk_widget_set_visible (GTK_WIDGET (self->shortcut_reset_label), FALSE);
+  gtk_widget_set_visible (GTK_WIDGET (self->shortcut_conflict_label), FALSE);
   self->edited = FALSE;
   setup_custom_shortcut (self);
 }
@@ -671,6 +674,7 @@ setup_keyboard_item (CcKeyboardShortcutEditor *self,
       g_signal_handlers_unblock_by_func (self->name_entry, name_entry_changed_cb, self);
     }
 
+  gtk_widget_set_visible (GTK_WIDGET (self->shortcut_reset_label), TRUE);
   g_free (accel);
   g_free (text);
 
@@ -933,6 +937,7 @@ cc_keyboard_shortcut_editor_class_init (CcKeyboardShortcutEditorClass *klass)
   gtk_widget_class_bind_template_child (widget_class, CcKeyboardShortcutEditor, reset_custom_button);
   gtk_widget_class_bind_template_child (widget_class, CcKeyboardShortcutEditor, set_button);
   gtk_widget_class_bind_template_child (widget_class, CcKeyboardShortcutEditor, shortcut_accel_label);
+  gtk_widget_class_bind_template_child (widget_class, CcKeyboardShortcutEditor, shortcut_reset_label);
   gtk_widget_class_bind_template_child (widget_class, CcKeyboardShortcutEditor, shortcut_conflict_label);
   gtk_widget_class_bind_template_child (widget_class, CcKeyboardShortcutEditor, standard_shortcut_stack);
   gtk_widget_class_bind_template_child (widget_class, CcKeyboardShortcutEditor, stack);
diff --git a/panels/keyboard/shortcut-editor.ui b/panels/keyboard/shortcut-editor.ui
index 81b931b..352825c 100644
--- a/panels/keyboard/shortcut-editor.ui
+++ b/panels/keyboard/shortcut-editor.ui
@@ -117,6 +117,19 @@
                             </child>
                           </object>
                         </child>
+                        <child>
+                          <object class="GtkLabel" id="shortcut_reset_label">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="label" translatable="yes">Press reset button on the right to change the command to default setting. Close current window to enter another command.</property>
+                            <property name="hexpand">True</property>
+                            <property name="wrap">True</property>
+                            <property name="wrap_mode">word-char</property>
+                            <property name="width_chars">15</property>
+                            <property name="max_width_chars">20</property>
+                            <property name="xalign">0</property>
+                          </object>
+                        </child>
                         <child>
                           <object class="GtkLabel" id="shortcut_conflict_label">
                             <property name="visible">True</property>
diff --git a/po/ko.po b/po/ko.po
index e196722..c47ae7d 100644
--- a/po/ko.po
+++ b/po/ko.po
@@ -1885,7 +1885,7 @@ msgstr ""
 "은 되돌릴 수 없습니다."
 
 #: panels/keyboard/cc-keyboard-panel.c:186
-#: panels/keyboard/shortcut-editor.ui:346
+#: panels/keyboard/shortcut-editor.ui:359
 #: panels/printers/authentication-dialog.ui:29
 #: panels/printers/ppd-selection-dialog.ui:27
 msgid "Cancel"
@@ -1974,45 +1974,53 @@ msgstr ""
 "키보드 바로 가기 입력에서 Esc를 누르면 취소하고, 백스페이스를 누르면 사용하지 않습"
 "니다."
 
-#: panels/keyboard/shortcut-editor.ui:156 panels/printers/details-dialog.ui:38
+#: panels/keyboard/shortcut-editor.ui:124
+msgid ""
+"Press reset button on the right to change the command to default setting. "
+"Close current window to enter another command."
+msgstr ""
+"단축키를 기본 설정으로 변경하려면 우측 초기화 버튼을 눌러야 합니다. "
+"다른 단축키를 입력하려면 현재 창을 닫아야 합니다"
+
+#: panels/keyboard/shortcut-editor.ui:169 panels/printers/details-dialog.ui:38
 #: panels/sound/gvc-mixer-dialog.c:1480
 #: panels/sound/gvc-sound-theme-chooser.c:554
 msgid "Name"
 msgstr "이름"
 
-#: panels/keyboard/shortcut-editor.ui:168
+#: panels/keyboard/shortcut-editor.ui:181
 msgid "Command"
 msgstr "명령"
 
-#: panels/keyboard/shortcut-editor.ui:180
+#: panels/keyboard/shortcut-editor.ui:193
 msgid "Shortcut"
 msgstr "바로 가기"
 
-#: panels/keyboard/shortcut-editor.ui:259
+#: panels/keyboard/shortcut-editor.ui:272
 msgid "Set Shortcut…"
 msgstr "바로 가기 설정…"
 
-#: panels/keyboard/shortcut-editor.ui:272 panels/network/network-wifi.ui:594
+#: panels/keyboard/shortcut-editor.ui:285 panels/network/network-wifi.ui:594
 msgid "None"
 msgstr "없음"
 
-#: panels/keyboard/shortcut-editor.ui:303
+#: panels/keyboard/shortcut-editor.ui:316
 msgid "Enter the new shortcut"
 msgstr "새 바로 가기 입력"
 
-#: panels/keyboard/shortcut-editor.ui:357
+#: panels/keyboard/shortcut-editor.ui:370
 msgid "Remove"
 msgstr "제거"
 
-#: panels/keyboard/shortcut-editor.ui:367
+#: panels/keyboard/shortcut-editor.ui:380
 msgid "Add"
 msgstr "추가"
 
-#: panels/keyboard/shortcut-editor.ui:382
+#: panels/keyboard/shortcut-editor.ui:395
 msgid "Replace"
 msgstr "바꾸기"
 
-#: panels/keyboard/shortcut-editor.ui:395
+#: panels/keyboard/shortcut-editor.ui:408
 msgid "Set"
 msgstr "설정"
 
