diff --git a/panels/user-accounts/data/account-dialog.ui b/panels/user-accounts/data/account-dialog.ui
index 9fd176f..d856d20 100644
--- a/panels/user-accounts/data/account-dialog.ui
+++ b/panels/user-accounts/data/account-dialog.ui
@@ -110,6 +110,21 @@
                     <property name="top_attach">2</property>
                   </packing>
                 </child>
+                <child>
+                  <object class="GtkCheckButton" id="home_directory_encrypt">
+                    <property name="label" translatable="yes">Home directory _Encryption</property>
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="receives_default">False</property>
+                    <property name="tooltip_text" translatable="yes">Encrypt all files in user's home directory with user's login password to protect sensitive data from people who can access to user's computer.</property>
+                    <property name="use_underline">True</property>
+                    <property name="draw_indicator">True</property>
+                  </object>
+                  <packing>
+                    <property name="left_attach">1</property>
+                    <property name="top_attach">4</property>
+                  </packing>
+                </child>
                 <child>
                   <object class="GtkLabel" id="local_username_hint">
                     <property name="visible">True</property>
@@ -119,7 +134,7 @@
                     <property name="label"></property>
                     <property name="width-chars">35</property>
                     <property name="max-width-chars">35</property>
-                    <property name="height-request">50</property>
+                    <property name="height-request">15</property>
                     <property name="wrap">True</property>
                     <property name="wrap_mode">word-char</property>
                     <property name="hexpand">True</property>
@@ -247,7 +262,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
-                    <property name="top_attach">4</property>
+                    <property name="top_attach">5</property>
                   </packing>
                 </child>
                 <child>
@@ -287,7 +302,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
-                    <property name="top_attach">5</property>
+                    <property name="top_attach">6</property>
                     <property name="width">2</property>
                     <property name="height">1</property>
                   </packing>
@@ -307,7 +322,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
-                    <property name="top_attach">6</property>
+                    <property name="top_attach">7</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -326,7 +341,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
-                    <property name="top_attach">6</property>
+                    <property name="top_attach">7</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -347,7 +362,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
-                    <property name="top_attach">7</property>
+                    <property name="top_attach">8</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -374,7 +389,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
-                    <property name="top_attach">8</property>
+                    <property name="top_attach">9</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -394,7 +409,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
-                    <property name="top_attach">9</property>
+                    <property name="top_attach">10</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -411,7 +426,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
-                    <property name="top_attach">9</property>
+                    <property name="top_attach">10</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -437,7 +452,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
-                    <property name="top_attach">10</property>
+                    <property name="top_attach">11</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -546,7 +561,7 @@
                     <property name="label"></property>
                     <property name="width-chars">35</property>
                     <property name="max-width-chars">35</property>
-                    <property name="height-request">50</property>
+                    <property name="height-request">15</property>
                     <property name="wrap">True</property>
                     <property name="wrap_mode">word-char</property>
                     <property name="hexpand">True</property>
@@ -642,7 +657,7 @@
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
-                    <property name="top_attach">4</property>
+                    <property name="top_attach">5</property>
                     <property name="width">1</property>
                     <property name="height">1</property>
                   </packing>
@@ -798,6 +813,7 @@
   </object>
   <object class="GtkSizeGroup" id="controls_sizegroup">
     <widgets>
+      <widget name="home_directory_encrypt"/>
       <widget name="local_username"/>
       <widget name="local_username_hint"/>
       <widget name="local_name"/>
diff --git a/panels/user-accounts/data/user-accounts-dialog.ui b/panels/user-accounts/data/user-accounts-dialog.ui
index 01ae11e..530f1ab 100644
--- a/panels/user-accounts/data/user-accounts-dialog.ui
+++ b/panels/user-accounts/data/user-accounts-dialog.ui
@@ -420,6 +420,26 @@
                     <property name="height">1</property>
                   </packing>
                 </child>
+                <child>
+                  <object class="GtkLabel" id="encrypt-home-check-label">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <property name="yalign">0</property>
+                    <property name="xalign">0</property>
+                    <property name="label" translatable="yes">Applied Home Directory Encryption.</property>
+                    <property name="height-request">15</property>
+                    <property name="wrap">True</property>
+                    <property name="wrap_mode">word-char</property>
+                    <property name="hexpand">True</property>
+                    <style>
+                      <class name="dim-label"/>
+                    </style>
+                  </object>
+                  <packing>
+                    <property name="left_attach">1</property>
+                    <property name="top_attach">7</property>
+                  </packing>
+                </child>
               </object>
             </child>
             <child>
@@ -510,6 +530,7 @@
       <widget name="account-type-box"/>
       <widget name="autologin-box"/>
       <widget name="last-login-button"/>
+      <widget name="encrypt-home-check-label"/>
     </widgets>
   </object>
   <object class="GtkSizeGroup">
diff --git a/panels/user-accounts/um-account-dialog.c b/panels/user-accounts/um-account-dialog.c
index 55bf5bc..a210d59 100644
--- a/panels/user-accounts/um-account-dialog.c
+++ b/panels/user-accounts/um-account-dialog.c
@@ -94,6 +94,7 @@ struct _UmAccountDialog {
         GtkWidget *local_strength_indicator;
         GtkWidget *local_hint;
         GtkWidget *local_verify_hint;
+        GtkWidget *home_directory_encrypt;
 
         /* Enterprise widgets */
         guint realmd_watch;
@@ -221,8 +222,9 @@ create_user_done (ActUserManager  *manager,
         if (user == NULL) {
                 finish_action (self);
                 g_debug ("Failed to create user: %s", error->message);
-                if (!g_error_matches (error, ACT_USER_MANAGER_ERROR, ACT_USER_MANAGER_ERROR_PERMISSION_DENIED))
+                if (!g_error_matches (error, ACT_USER_MANAGER_ERROR, ACT_USER_MANAGER_ERROR_PERMISSION_DENIED)) {
                        show_error_dialog (self, _("Failed to add account"), error);
+        }
                 g_error_free (error);
                 gtk_widget_grab_focus (self->local_name);
         } else {
@@ -243,20 +245,23 @@ local_create_user (UmAccountDialog *self)
         const gchar *username;
         const gchar *name;
         gint account_type;
+        gboolean encrypt_home;
 
         begin_action (self);
 
         name = gtk_entry_get_text (GTK_ENTRY (self->local_name));
         username = gtk_combo_box_text_get_active_text (GTK_COMBO_BOX_TEXT (self->local_username));
         account_type = (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (self->account_type_standard)) ? ACT_USER_ACCOUNT_TYPE_STANDARD : ACT_USER_ACCOUNT_TYPE_ADMINISTRATOR);
+        encrypt_home = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (self->home_directory_encrypt));
 
         g_debug ("Creating local user: %s", username);
 
-        manager = act_user_manager_get_default ();
-        act_user_manager_create_user_async (manager,
+        manager = act_user_manager_get_default ();
+        act_user_manager_create_user_encrypt_async (manager,
                                             g_ascii_strdown (username, -1),
                                             name,
                                             account_type,
+                                            encrypt_home,
                                             self->cancellable,
                                             (GAsyncReadyCallback)create_user_done,
                                             self);
@@ -1594,6 +1599,7 @@ um_account_dialog_class_init (UmAccountDialogClass *klass)
         gtk_widget_class_bind_template_child (widget_class, UmAccountDialog, local_strength_indicator);
         gtk_widget_class_bind_template_child (widget_class, UmAccountDialog, local_hint);
         gtk_widget_class_bind_template_child (widget_class, UmAccountDialog, local_verify_hint);
+        gtk_widget_class_bind_template_child (widget_class, UmAccountDialog, home_directory_encrypt);
         gtk_widget_class_bind_template_child (widget_class, UmAccountDialog, enterprise_button);
         gtk_widget_class_bind_template_child (widget_class, UmAccountDialog, spinner);
         gtk_widget_class_bind_template_child (widget_class, UmAccountDialog, enterprise_domain);
diff --git a/panels/user-accounts/um-user-panel.c b/panels/user-accounts/um-user-panel.c
index 35b3b52..26d66bd 100644
--- a/panels/user-accounts/um-user-panel.c
+++ b/panels/user-accounts/um-user-panel.c
@@ -86,6 +86,8 @@ struct _CcUserPanel {
 
 CC_PANEL_REGISTER (CcUserPanel, cc_user_panel)
 
+static gboolean is_ecryptfs_user (const gchar *user);
+
 static GtkWidget *
 get_widget (CcUserPanel *self, const char *name)
 {
@@ -399,7 +401,7 @@ delete_user_response (GtkWidget   *dialog,
         else if (response_id == GTK_RESPONSE_NO) {
                 remove_files = TRUE;
         }
-        else {
+        else {
                 remove_files = FALSE;
         }
 
@@ -616,21 +618,37 @@ delete_user (GtkButton *button, CcUserPanel *self)
                                   G_CALLBACK (gtk_widget_destroy), NULL);
         }
         else if (act_user_is_local_account (user)) {
-                dialog = gtk_message_dialog_new (GTK_WINDOW (gtk_widget_get_toplevel (self->main_box)),
-                                                 0,
-                                                 GTK_MESSAGE_QUESTION,
-                                                 GTK_BUTTONS_NONE,
-                                                 _("Do you want to keep %s’s files?"),
-                                                get_real_or_user_name (user));
-
-                gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog),
-                                                          _("It is possible to keep the home directory, mail spool and temporary files around when deleting a user account."));
-
-                gtk_dialog_add_buttons (GTK_DIALOG (dialog),
-                                        _("_Delete Files"), GTK_RESPONSE_NO,
-                                        _("_Keep Files"), GTK_RESPONSE_YES,
-                                        _("_Cancel"), GTK_RESPONSE_CANCEL,
-                                        NULL);
+                if (is_ecryptfs_user (act_user_get_user_name (user))) {
+                        dialog = gtk_message_dialog_new (GTK_WINDOW (gtk_widget_get_toplevel (self->main_box)),
+                                                         0,
+                                                         GTK_MESSAGE_QUESTION,
+                                                         GTK_BUTTONS_NONE,
+                                                         _("Do you want to delete user %s?"),
+                                                         get_real_or_user_name (user));
+
+                        gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog),
+                                                                  _("All data related with user %s will be deleted."),get_real_or_user_name(user));
+
+                        gtk_dialog_add_buttons (GTK_DIALOG (dialog),
+                                                _("_Delete Files"), GTK_RESPONSE_NO,
+                                                _("_Cancel"), GTK_RESPONSE_CANCEL,
+                                                NULL);
+                } else {
+                        dialog = gtk_message_dialog_new (GTK_WINDOW (gtk_widget_get_toplevel (self->main_box)),
+                                                         0,
+                                                         GTK_MESSAGE_QUESTION,
+                                                         GTK_BUTTONS_NONE,
+                                                         _("Do you want to keep %s’s files?"),
+                                                         get_real_or_user_name (user));
+
+                        gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog),
+                                                                  _("It is possible to keep the home directory, mail spool and temporary files around when deleting a user account."));
+                        gtk_dialog_add_buttons (GTK_DIALOG (dialog),
+                                                _("_Delete Files"), GTK_RESPONSE_NO,
+                                                _("_Keep Files"), GTK_RESPONSE_YES,
+                                                _("_Cancel"), GTK_RESPONSE_CANCEL,
+                                                NULL);
+                }
 
                 gtk_window_set_icon_name (GTK_WINDOW (dialog), "system-users");
 
@@ -888,6 +906,8 @@ show_user (ActUser *user, CcUserPanel *self)
         show = is_ecryptfs_user (act_user_get_user_name (user));//act_user_is_local_account (user);
         gtk_widget_set_visible (widget, !show);
         gtk_widget_set_visible (label, !show);
+        label = get_widget (self, "encrypt-home-check-label");
+        gtk_widget_set_visible (label, show);
 
         /* Language: do not show for current user */
         widget = get_widget (self, "account-language-button");
diff --git a/po/ko.po b/po/ko.po
index 5324be0..e196722 100644
--- a/po/ko.po
+++ b/po/ko.po
@@ -144,8 +144,9 @@ msgstr "이미지 파일을 %s 폴더에 놓으면 여기 나타납니다"
 #: panels/user-accounts/um-fingerprint-dialog.c:261
 #: panels/user-accounts/um-photo-dialog.c:103
 #: panels/user-accounts/um-photo-dialog.c:230
-#: panels/user-accounts/um-user-panel.c:631
+#: panels/user-accounts/um-user-panel.c:634
 #: panels/user-accounts/um-user-panel.c:649
+#: panels/user-accounts/um-user-panel.c:668
 msgid "_Cancel"
 msgstr "취소(_C)"
 
@@ -3474,8 +3475,8 @@ msgstr "EAP-LEAP 암호가 없습니다"
 #: panels/network/wireless-security/eap-method-leap.ui:26
 #: panels/network/wireless-security/eap-method-simple.ui:26
 #: panels/network/wireless-security/ws-leap.ui:26
-#: panels/user-accounts/data/account-dialog.ui:142
-#: panels/user-accounts/data/account-dialog.ui:505
+#: panels/user-accounts/data/account-dialog.ui:157
+#: panels/user-accounts/data/account-dialog.ui:520
 msgid "_Username"
 msgstr "사용자 이름(_U)"
 
@@ -4312,7 +4313,7 @@ msgstr "사용자 이름"
 #. Translators: This is a password needed for printing.
 #: panels/printers/authentication-dialog.ui:96
 #: panels/printers/jobs-dialog.ui:70 panels/printers/new-printer-dialog.ui:383
-#: panels/user-accounts/data/account-dialog.ui:240
+#: panels/user-accounts/data/account-dialog.ui:255
 msgid "Password"
 msgstr "암호"
 
@@ -6558,39 +6559,47 @@ msgstr "색상 효과"
 msgid "Add User"
 msgstr "사용자 추가"
 
-#: panels/user-accounts/data/account-dialog.ui:171
+#: panels/user-accounts/data/account-dialog.ui:115
+msgid "Home directory _Encryption"
+msgstr "홈폴더 암호화(_E)"
+
+#: panels/user-accounts/data/account-dialog.ui:119
+msgid "Encrypt all files in user's home directory with user's login password to protect sensitive da     ta from people who can access to user's computer."
+msgstr "사용자의 홈폴더에 있는 모든 파일을 사용자의 로그인 암호로 암호화하여 사용자의 컴퓨터에 물리>     적으로 접근 가능한 사람으로부터 민감한 정보를 보호합니다."
+
+#: panels/user-accounts/data/account-dialog.ui:186
 msgid "_Full Name"
 msgstr "전체 이름(_F)"
 
-#: panels/user-accounts/data/account-dialog.ui:197
+#: panels/user-accounts/data/account-dialog.ui:212
 #: panels/user-accounts/data/user-accounts-dialog.ui:146
 msgid "Standard"
 msgstr "표준"
 
-#: panels/user-accounts/data/account-dialog.ui:207
+#: panels/user-accounts/data/account-dialog.ui:222
 #: panels/user-accounts/data/user-accounts-dialog.ui:155
 msgid "Administrator"
 msgstr "관리자"
 
-#: panels/user-accounts/data/account-dialog.ui:223
+#: panels/user-accounts/data/account-dialog.ui:238
 #: panels/user-accounts/data/user-accounts-dialog.ui:173
 msgid "Account _Type"
 msgstr "계정 종류(_T)"
 
-#: panels/user-accounts/data/account-dialog.ui:260
+#: panels/user-accounts/data/account-dialog.ui:275
 msgid "Allow user to set a password when they next _login"
 msgstr "다음 로그인에 암호를 설정하도록 허용(_L)"
 
-#: panels/user-accounts/data/account-dialog.ui:274
+#: panels/user-accounts/data/account-dialog.ui:289
 msgid "Set a password _now"
 msgstr "지금 암호 설정(_N)"
 
-#: panels/user-accounts/data/account-dialog.ui:387
+#: panels/user-accounts/data/account-dialog.ui:402
 msgid "_Confirm"
 msgstr "확인(_C)"
 
-#: panels/user-accounts/data/account-dialog.ui:463
-#: panels/user-accounts/data/account-dialog.ui:667
+#: panels/user-accounts/data/account-dialog.ui:478
+#: panels/user-accounts/data/account-dialog.ui:682
 msgid ""
 "Enterprise login allows an existing centrally managed user account to be "
 "used on this device. You can also use this account to access company "
@@ -6599,20 +6608,20 @@ msgstr ""
 "기업 로그인에서는 중앙에서 관리하는 사용자 계정을 이 장치에서 사용합니다. 이 "
 "계정을 사용해서 인터넷에서 회사의 자원에 접근할 수 잇습니다."
 
-#: panels/user-accounts/data/account-dialog.ui:485
+#: panels/user-accounts/data/account-dialog.ui:500
 #: panels/user-accounts/data/join-dialog.ui:116
 msgid "_Domain"
 msgstr "도메인(_D)"
 
-#: panels/user-accounts/data/account-dialog.ui:707
+#: panels/user-accounts/data/account-dialog.ui:722
 msgid "You are Offline"
 msgstr "연결이 끊겼습니다"
 
-#: panels/user-accounts/data/account-dialog.ui:726
+#: panels/user-accounts/data/account-dialog.ui:741
 msgid "You must be online in order to add enterprise users."
 msgstr "기업 사용자를 추가하려면 연결된 상태여야 합니다."
 
-#: panels/user-accounts/data/account-dialog.ui:760
+#: panels/user-accounts/data/account-dialog.ui:775
 msgid "_Enterprise Login"
 msgstr "기업 로그인(_E)"
 
@@ -6785,16 +6794,20 @@ msgstr "사용자 아이콘"
 msgid "Last Login"
 msgstr "최근 로그인"
 
-#: panels/user-accounts/data/user-accounts-dialog.ui:434
+#: panels/user-accounts/data/user-accounts-dialog.ui:429
+msgid "Applied Home Directory Encryption."
+msgstr "홈폴더 암호화가 적용되어 있습니다."
+
+#: panels/user-accounts/data/user-accounts-dialog.ui:456
 msgid "Remove User…"
 msgstr "사용자 제거…"
 
 #. Translators: This is the empty state page label which states that there are no users to show in the panel.
-#: panels/user-accounts/data/user-accounts-dialog.ui:466
+#: panels/user-accounts/data/user-accounts-dialog.ui:488
 msgid "No Users Found"
 msgstr "사용자가 없습니다"
 
-#: panels/user-accounts/data/user-accounts-dialog.ui:476
+#: panels/user-accounts/data/user-accounts-dialog.ui:498
 msgid "Unlock to add a user account."
 msgstr "사용자 계정을 추가하려면 잠금을 푸십시오."
 
@@ -6962,34 +6975,34 @@ msgstr "알 수 없는 오류"
 msgid "Should match the web address of your login provider."
 msgstr "로그인을 제공하는 곳의 웹 주소."
 
-#: panels/user-accounts/um-account-dialog.c:140
+#: panels/user-accounts/um-account-dialog.c:141
 msgid "Username cannot start with digits."
 msgstr "사용자 이름은 숫자로 시작할 수 없습니다."
 
-#: panels/user-accounts/um-account-dialog.c:225
+#: panels/user-accounts/um-account-dialog.c:226
 msgid "Failed to add account"
 msgstr "계정 추가에 실패했습니다"
 
-#: panels/user-accounts/um-account-dialog.c:465
+#: panels/user-accounts/um-account-dialog.c:470
 #: panels/user-accounts/um-password-dialog.c:291
 msgid "The passwords do not match."
 msgstr "암호가 다릅니다."
 
-#: panels/user-accounts/um-account-dialog.c:736
-#: panels/user-accounts/um-account-dialog.c:782
-#: panels/user-accounts/um-account-dialog.c:803
+#: panels/user-accounts/um-account-dialog.c:741
+#: panels/user-accounts/um-account-dialog.c:787
+#: panels/user-accounts/um-account-dialog.c:808
 msgid "Failed to register account"
 msgstr "계정 등록에 실패했습니다"
 
-#: panels/user-accounts/um-account-dialog.c:926
+#: panels/user-accounts/um-account-dialog.c:931
 msgid "No supported way to authenticate with this domain"
 msgstr "이 도메인에 대해 인증하기 위한 방법을 지원하지 않습니다"
 
-#: panels/user-accounts/um-account-dialog.c:999
+#: panels/user-accounts/um-account-dialog.c:1004
 msgid "Failed to join domain"
 msgstr "도메인 참여에 실패했습니다"
 
-#: panels/user-accounts/um-account-dialog.c:1060
+#: panels/user-accounts/um-account-dialog.c:1065
 msgid ""
 "That login name didn’t work.\n"
 "Please try again."
@@ -6997,7 +7010,7 @@ msgstr ""
 "이 로그인 이름은 쓸 수 없습니다.\n"
 "다시 시도해 보십시오."
 
-#: panels/user-accounts/um-account-dialog.c:1067
+#: panels/user-accounts/um-account-dialog.c:1072
 msgid ""
 "That login password didn’t work.\n"
 "Please try again."
@@ -7005,11 +7018,11 @@ msgstr ""
 "이 로그인 암호는 쓸 수 없습니다.\n"
 "다시 시도해 보십시오."
 
-#: panels/user-accounts/um-account-dialog.c:1075
+#: panels/user-accounts/um-account-dialog.c:1080
 msgid "Failed to log into domain"
 msgstr "도메인에 로그를 남기는데 실패했습니다"
 
-#: panels/user-accounts/um-account-dialog.c:1133
+#: panels/user-accounts/um-account-dialog.c:1138
 msgid "Unable to find the domain. Maybe you misspelled it?"
 msgstr "도메인을 찾을 수 없습니다. 잘못 쓰신 것이 아닙니까?"
 
@@ -7139,7 +7152,7 @@ msgstr "%s — %s"
 #. Translators: This is a time format string in the style of "22:58".
 #. It indicates a login time which follows a date.
 #: panels/user-accounts/um-history-dialog.c:177
-#: panels/user-accounts/um-user-panel.c:763
+#: panels/user-accounts/um-user-panel.c:782
 msgctxt "login date-time"
 msgid "%k:%M"
 msgstr "%k:%M"
@@ -7147,7 +7160,7 @@ msgstr "%k:%M"
 #. Translators: This indicates a login date-time.
 #. The first %s is a date, and the second %s a time.
 #: panels/user-accounts/um-history-dialog.c:180
-#: panels/user-accounts/um-user-panel.c:767
+#: panels/user-accounts/um-user-panel.c:786
 #, c-format
 msgctxt "login date-time"
 msgid "%s, %s"
@@ -7207,41 +7220,56 @@ msgstr "암호가 잘못되었으니 다시 시도해주십시오"
 msgid "Couldn’t connect to the %s domain: %s"
 msgstr "%s 도메인에 연결할 수 없습니다: %s"
 
-#: panels/user-accounts/um-user-panel.c:199
+#: panels/user-accounts/um-user-panel.c:202
 msgid "Your account"
 msgstr "내 계정"
 
-#: panels/user-accounts/um-user-panel.c:379
+#: panels/user-accounts/um-user-panel.c:382
 msgid "Failed to delete user"
 msgstr "사용자 삭제에 실패했습니다"
 
-#: panels/user-accounts/um-user-panel.c:437
-#: panels/user-accounts/um-user-panel.c:496
-#: panels/user-accounts/um-user-panel.c:548
+#: panels/user-accounts/um-user-panel.c:440
+#: panels/user-accounts/um-user-panel.c:499
+#: panels/user-accounts/um-user-panel.c:551
 msgid "Failed to revoke remotely managed user"
 msgstr "원격에서 관리하는 사용자를 철회하는데 실패했습니다"
 
-#: panels/user-accounts/um-user-panel.c:600
+#: panels/user-accounts/um-user-panel.c:603
 msgid "You cannot delete your own account."
 msgstr "자기 계정은 삭제할 수 없습니다."
 
-#: panels/user-accounts/um-user-panel.c:609
+#: panels/user-accounts/um-user-panel.c:612
 #, c-format
 msgid "%s is still logged in"
 msgstr "%s 사용자가 아직 로그인 중입니다"
 
-#: panels/user-accounts/um-user-panel.c:613
+#: panels/user-accounts/um-user-panel.c:616
 msgid ""
 "Deleting a user while they are logged in can leave the system in an "
 "inconsistent state."
 msgstr "로그인 중인 사용자를 삭제하면 시스템이 불안정한 상태가 될 수 있습니다."
 
-#: panels/user-accounts/um-user-panel.c:622
+#: panels/user-accounts/um-user-panel.c:626
+msgid "Do you want to delete user %s?"
+msgstr "정말로 %s 사용자를 삭제하시겠습니까?"
+
+#: panels/user-accounts/um-user-panel.c:630
+msgid "All data related with user %s will be deleted."
+msgstr ""
+"%s 사용자와 관련된 모든 데이터가 삭제됩니다. 홈폴더 암호화가 적용된 계정은"
+"계정 삭제시 항상 모든 데이터를 삭제합니다."
+
+#: panels/user-accounts/um-user-panel.c:633
+#: panels/user-accounts/um-user-panel.c:647
+msgid "_Delete Files"
+msgstr "파일 삭제(_D)"
+
+#: panels/user-accounts/um-user-panel.c:641
 #, c-format
 msgid "Do you want to keep %s’s files?"
 msgstr "%s 사용자의 파일을 유지하시겠습니까?"
 
-#: panels/user-accounts/um-user-panel.c:626
+#: panels/user-accounts/um-user-panel.c:645
 msgid ""
 "It is possible to keep the home directory, mail spool and temporary files "
 "around when deleting a user account."
@@ -7249,47 +7277,43 @@ msgstr ""
 "사용자 계정을 삭제할 때 홈 디렉터리, 메일 스풀, 임시 파일을 유지할 수 있습니"
 "다."
 
-#: panels/user-accounts/um-user-panel.c:629
-msgid "_Delete Files"
-msgstr "파일 삭제(_D)"
-
-#: panels/user-accounts/um-user-panel.c:630
+#: panels/user-accounts/um-user-panel.c:648
 msgid "_Keep Files"
 msgstr "파일 유지(_K)"
 
-#: panels/user-accounts/um-user-panel.c:644
+#: panels/user-accounts/um-user-panel.c:663
 #, c-format
 msgid "Are you sure you want to revoke remotely managed %s’s account?"
 msgstr "정말로 원격에서 관리하는 %s의 계정을 철회하시겠습니까?"
 
-#: panels/user-accounts/um-user-panel.c:648
+#: panels/user-accounts/um-user-panel.c:667
 msgid "_Delete"
 msgstr "삭제(_D)"
 
-#: panels/user-accounts/um-user-panel.c:698
+#: panels/user-accounts/um-user-panel.c:717
 msgctxt "Password mode"
 msgid "Account disabled"
 msgstr "계정 사용 중지됨"
 
-#: panels/user-accounts/um-user-panel.c:706
+#: panels/user-accounts/um-user-panel.c:725
 msgctxt "Password mode"
 msgid "To be set at next login"
 msgstr "다음 로그인에 설정 예정"
 
-#: panels/user-accounts/um-user-panel.c:709
+#: panels/user-accounts/um-user-panel.c:728
 msgctxt "Password mode"
 msgid "None"
 msgstr "없음"
 
-#: panels/user-accounts/um-user-panel.c:756
+#: panels/user-accounts/um-user-panel.c:775
 msgid "Logged in"
 msgstr "로그인함"
 
-#: panels/user-accounts/um-user-panel.c:1105
+#: panels/user-accounts/um-user-panel.c:1169
 msgid "Failed to contact the accounts service"
 msgstr "계정 서비스 연결에 실패했습니다"
 
-#: panels/user-accounts/um-user-panel.c:1107
+#: panels/user-accounts/um-user-panel.c:1171
 msgid "Please make sure that the AccountService is installed and enabled."
 msgstr "AccountService를 설치하고 사용 표시했는지 확인하십시오."
 
@@ -7297,7 +7321,7 @@ msgstr "AccountService를 설치하고 사용 표시했는지 확인하십시오
 #. * We split the line in 2 here to "make it look good", as there's
 #. * no good way to do this in GTK+ for tooltips. See:
 #. * https://bugzilla.gnome.org/show_bug.cgi?id=657168
-#: panels/user-accounts/um-user-panel.c:1139
+#: panels/user-accounts/um-user-panel.c:1203
 msgid ""
 "To make changes,\n"
 "click the * icon first"
@@ -7305,12 +7329,12 @@ msgstr ""
 "바꾸려면,\n"
 "먼저 * 아이콘을 누르십시오"
 
-#: panels/user-accounts/um-user-panel.c:1179
+#: panels/user-accounts/um-user-panel.c:1241
 msgid "Create a user account"
 msgstr "새 사용자 계정 만들기"
 
-#: panels/user-accounts/um-user-panel.c:1190
-#: panels/user-accounts/um-user-panel.c:1368
+#: panels/user-accounts/um-user-panel.c:1254
+#: panels/user-accounts/um-user-panel.c:1432
 msgid ""
 "To create a user account,\n"
 "click the * icon first"
@@ -7318,12 +7342,12 @@ msgstr ""
 "사용자 계정을 만드려면,\n"
 "먼저 * 아이콘을 누르십시오"
 
-#: panels/user-accounts/um-user-panel.c:1200
+#: panels/user-accounts/um-user-panel.c:1264
 msgid "Delete the selected user account"
 msgstr "선택한 사용자 계정 삭제"
 
-#: panels/user-accounts/um-user-panel.c:1212
-#: panels/user-accounts/um-user-panel.c:1373
+#: panels/user-accounts/um-user-panel.c:1276
+#: panels/user-accounts/um-user-panel.c:1437
 msgid ""
 "To delete the selected user account,\n"
 "click the * icon first"
@@ -7890,7 +7914,7 @@ msgstr ""
 "매체제어 접근 권한이 없습니다.\n"
 "사용하시려면 접근 권한을 \"allow\"로 변경하십시오."
 
-#: ../panels/user-accounts/um-user-panel.c:997
+#: ../panels/user-accounts/um-user-panel.c:1040
 msgid ""
 "User infomations has changed.\n"
 "Changes take effect after your session needs to be restarted.\n"
